packages/select/src/Option.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useCallback, useContext, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { usePrevious } from '@leafygreen-ui/hooks';
+import { LGGlyph } from '@leafygreen-ui/icon/src/types';
+import { colorSets } from './styleSets';
+import SelectContext from './SelectContext';
+
+type GlyphElement = React.ReactElement<LGGlyph.ComponentProps> & {
```

Comment:
Should we move this type to the Icon directory, can imagine it'd be useful across the repo

Reply:
I think it'd be good to do. hoping to keep it as a follow-up to try to keep this PR more self-contained.

Reply:
makes sense to me